CRM/Member/Form/Search.php

```diff
@@ -305,8 +305,13 @@ public function fixFormValues() {
     $membershipType = CRM_Utils_Request::retrieve('type', 'String');
 
     if ($membershipType) {
-      $this->_formValues['membership_type_id'] = array($membershipType);
-      $this->_defaults['membership_type_id'] = array($membershipType);
+			$typearray=explode(",",$membershipType);
+			$typeresult = array();
+			foreach ($typearray as $type){
+				$typeresult+=array($type => 1);
+			}
+			$this->_formValues['member_membership_type_id'] = $typeresult;
+			$this->_defaults['member_membership_type_id'] = $typeresult;
```

Comment:
Please indent with 2 spaces, no tabs, and a space after commas and surrounding operators.